node/avn-service/src/ethereum_events_handler.rs

```diff
@@ -315,6 +315,55 @@ pub async fn identify_events(
     Ok(unique_transactions.into_values().collect())
 }
 
+pub async fn identify_additional_events(
+    web3: &Web3<web3::transports::Http>,
+    start_block: u32,
+    contract_addresses: &Vec<H160>,
+    event_signatures_to_find: &Vec<SpH256>,
+    events_registry: &EventRegistry,
+    additional_events_to_check: Vec<AdditionalEvent>,
+) -> Result<Vec<DiscoveredEvent>, AppError> {
+    // Create a future for each event
+    let futures = additional_events_to_check
+        .iter()
+        .filter(|event| {
+            event.block < start_block &&
+                event_signatures_to_find.contains(&event.event_id.signature)
+        })
+        .map(|event| {
+            let contract = contract_addresses.clone();
```

Comment:
do we need to clone here?

Reply:
yes cause we have any async closure that we pass that in.